modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -112,7 +117,9 @@ class Instrument_Manager extends \NDB_Menu_Filter
      */
     protected function handlePOST(ServerRequestInterface $request): ResponseInterface
     {
-        if (!$this->canWriteFiles()) {
+        if (! ($this->canWriteFiles()
+            && \User::singleton()->hasPermission('instrument_manager_write'))
```

Comment:
I would put this check in a separated `if` statement because the error message won't be the same in the Forbidden response.
Also, split line expression in if statement always look weird. (But that is subjective)

Reply:
I agree with @xlecours.. it should be a different response status based on why, so it needs to be 2 different if statements to do that.